### PR TITLE
Feature: Add ALU XOR implementation

### PR DESCRIPTION
**Pull request by mormod**
... as well as unit tests. See p. 85.

### CODE REVIEW

File: test/alu_8bit_xor_test.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#include <criterion/new/assert.h>
+#include <criterion/parameterized.h>
+
+#include "fixtures/cpu_mmu.h"
+#include "common/util.h"
+#include "common/alu.h"
+
+ParameterizedTestParameters(XOR_A_n, XOR_A_n_same_destruction) {
+    const uint8_t alternate    = 0b10101010;
+    const uint8_t f_wanted     = 0x00 | (1 << Z_FLAG);
```
Comment (marcluque) [inline test/alu_8bit_xor_test.c:11]: Just a very tiny naming thing. Usually, these things are called `expected` like `expected_freg` or something. But I suppose this this preference and pretty much up to your personal preference.

Comment (marcluque) [reply test/alu_8bit_xor_test.c:11]: Apart from that: good idea to make It a bit more expressive like this.

I was thinking about making some defines that could be "ored" like `ZFLAG | CFLAG`. But I'm not sure how well they would go with the enum we currently have :/

**Review by marcluque — APPROVED** [review]
LGTM!

File: test/alu_8bit_xor_test.c (hunk 1 above)
Comment (mormod) [reply test/alu_8bit_xor_test.c:11]: Concerning the enum "concatenation": One could rename those values to, i.e., `Z_FLAG_POS` and introduce new defines, such as
```
#define Z_FLAG (1U << Z_FLAG_POS)
```
Then `uint8t_t flags = Z_FLAG | C_FLAG`  would work.

And __expected__ just didn't want to come to my mind, so I just used __wanted__ instead :D

Comment (marcluque) [reply test/alu_8bit_xor_test.c:11]: Ah nice, that's a nice idea, I'd suggest we do that.

Exactly, that seems a bit more concise and less cryptic to me than 0b... 😆

**Pull request merged**